apollo-router/src/query_planner/bridge_query_planner.rs

```diff
@@ -215,14 +215,17 @@ impl BridgeQueryPlanner {
         let selections = self.parse_selections(key.0.clone()).await?;
 
         if selections.contains_introspection() {
-            // If we have only one operation containing a single root field `__typename`
-            if selections.contains_only_typename() {
+            // If we have only one operation containing a only the root field `__typename`
+            // (possibly aliased or repeated)
+            if let Some(output_keys) = selections.contains_only_typenames_with_output_keys() {
+                let operation_name = selections.operations[0].kind().to_string();
```

Comment:
Not your change (since this already existed, but now I'm looking at it)

Seems a bit risky to assume that `operations` has at least 1 entry and could panic if it doesn't.

If we "know" that it must always have at least length 1, then we should add a comment to document this. Otherwise, consider using `first()` with pattern matching or verifying `!operations.is_empty()`

Reply:
This is actually correct today because both of the conditionals it's inside require non-empty operations. But I do understand that making that more clear at compile time could be better. It does seem like the fact that we have actively seen an operation with the given shape is relevant.  Maybe I should just get rid of the only-called-once contains_only_typenames_with_output_keys...

Reply:
OK, that's what I did. Note that I did have to expand visibility of `Operation::is_only_typenames_with_output_keys` to do so.